test/common/scale_up_replicas_threescale.go

```diff
@@ -0,0 +1,152 @@
+		t.Fatalf("Incorrect number of replicas : %v", err)
+	}
+
+	apim, err = updateAPIManager(ctx.Client, t, numberOfReplicas)
+	if err != nil {
+		t.Fatalf("Unable to update : %v", err)
+	}
+
+	t.Log("Checking correct number of replicas has been reset")
+	if err := checkNumberOfReplicasAgainstValue(apim, numberOfReplicas, createWorkspaceRetryInterval, createWorkspaceTimeout, t); err != nil {
+		t.Fatalf("Incorrect number of replicas : %v", err)
+	}
+}
+
+func getAPIManager(dynClient k8sclient.Client, t *testing.T) (threescalev1.APIManager, error) {
+	apim := &threescalev1.APIManager{}
+
+	if err := dynClient.Get(goctx.TODO(), types.NamespacedName{Name: name, Namespace: namespace}, apim); err != nil {
+		t.Fatalf("Error : %v ", err)
+	}
+
+	return *apim, nil
+}
+
+func updateAPIManager(dynClient k8sclient.Client, t *testing.T, replicas int64) (threescalev1.APIManager, error) {
+
+	apim, err := getAPIManager(dynClient, t)
+	if err != nil {
+		t.Fatalf("Error : %v ", err)
+	}
+	resourceRequirements := true
+	newApim := &threescalev1.APIManager{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            name,
+			Namespace:       namespace,
+			ResourceVersion: apim.GetResourceVersion(),
+		},
+		Spec: threescalev1.APIManagerSpec{
+			HighAvailability: &threescalev1.HighAvailabilitySpec{},
+			APIManagerCommonSpec: threescalev1.APIManagerCommonSpec{
+				ResourceRequirementsEnabled: &resourceRequirements,
+			},
+			System: &threescalev1.SystemSpec{
+				DatabaseSpec: &threescalev1.SystemDatabaseSpec{
+					PostgreSQL: &threescalev1.SystemPostgreSQLSpec{},
+				},
+				FileStorageSpec: &threescalev1.SystemFileStorageSpec{
+					S3: &threescalev1.SystemS3Spec{},
+				},
+				AppSpec:     &threescalev1.SystemAppSpec{Replicas: &[]int64{replicas}[0]},
+				SidekiqSpec: &threescalev1.SystemSidekiqSpec{Replicas: &[]int64{replicas}[0]},
+			},
+			Apicast: &threescalev1.ApicastSpec{
+				ProductionSpec: &threescalev1.ApicastProductionSpec{Replicas: &[]int64{replicas}[0]},
+				StagingSpec:    &threescalev1.ApicastStagingSpec{Replicas: &[]int64{replicas}[0]},
+			},
+			Backend: &threescalev1.BackendSpec{
+				ListenerSpec: &threescalev1.BackendListenerSpec{Replicas: &[]int64{replicas}[0]},
+				WorkerSpec:   &threescalev1.BackendWorkerSpec{Replicas: &[]int64{replicas}[0]},
+				CronSpec:     &threescalev1.BackendCronSpec{Replicas: &[]int64{replicas}[0]},
+			},
+			Zync: &threescalev1.ZyncSpec{
+				AppSpec: &threescalev1.ZyncAppSpec{Replicas: &[]int64{replicas}[0]},
+				QueSpec: &threescalev1.ZyncQueSpec{Replicas: &[]int64{replicas}[0]},
+			},
+		},
+	}
+
+	if err := dynClient.Update(goctx.TODO(), newApim, &k8sclient.UpdateOptions{}); err != nil {
+		t.Fatalf("Error : %v ", err)
+	}
+
+	return *newApim, nil
+}
+
+func checkNumberOfReplicasAgainstValue(apim threescalev1.APIManager, numberOfRequiredReplicas int64, retryInterval, timeout time.Duration, t *testing.T) error {
+	return wait.Poll(retryInterval, timeout, func() (done bool, err error) {
```

Comment:
The APIManager CR will need to be retrieved again from the cluster when polling to get the updated version.